private/pkg/transport/http2client/client_option.go

```diff
@@ -51,3 +61,13 @@ func WithObservability() ClientOption {
 		option.observability = true
 	}
 }
+
+// WithProxy returns a new ClientOption to use
+// a proxy.
+//
+// The default is to use http.ProxyFromEnvironment
+func WithProxy(proxyFunc Proxy) ClientOption {
+	return func(option *clientOption) {
+		option.proxy = proxyFunc
+	}
+}
```

Comment:
If we never use this option, why have it? I think we want the CLI and everything in core to use `ProxyFromEnvironment` - maybe we can add this later on, when we know it's necessary.